### ISSUE TITLE
 LibTorch make error

### ISSUE THREAD

**Issue by 3bobo** [pytorch/pytorch#12898]
cmake version 3.5.1
Make error:
```
-- The C compiler identification is GNU 5.4.0
-- The CXX compiler identification is GNU 5.4.0
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Looking for pthread_create
-- Looking for pthread_create - not found
-- Looking for pthread_create in pthreads
-- Looking for pthread_create in pthreads - not found
-- Looking for pthread_create in pthread
-- Looking for pthread_create in pthread - found
-- Found Threads: TRUE  
-- Found CUDA: /usr/local/cuda-9.0 (found suitable version "9.0", minimum required is "7.0") 
-- Caffe2: CUDA detected: 9.0
-- Caffe2: CUDA nvcc is: /usr/local/cuda-9.0/bin/nvcc
-- Caffe2: CUDA toolkit directory: /usr/local/cuda-9.0
-- Caffe2: Header version is: 9.0
-- Found CUDNN: /usr/local/cuda-9.0/include  
-- Found cuDNN: v7.1.2  (include: /usr/local/cuda-9.0/include, library: /usr/local/cuda-9.0/lib64/libcudnn.so)
CMake Error at /home/zqk/download/libtorch/share/cmake/Caffe2/public/utils.cmake:192 (cuda_select_nvcc_arch_flags):
  Unknown CMake command "cuda_select_nvcc_arch_flags".
Call Stack (most recent call first):
  /home/zqk/download/libtorch/share/cmake/Caffe2/public/cuda.cmake:332 (torch_cuda_get_nvcc_gencode_flag)
  /home/zqk/download/libtorch/share/cmake/Caffe2/Caffe2Config.cmake:88 (include)
  /home/zqk/download/libtorch/share/cmake/Torch/TorchConfig.cmake:37 (find_package)
  CMakeLists.txt:4 (find_package)


-- Configuring incomplete, errors occurred!
See also "/home/zqk/work/project/deep&ros/build/CMakeFiles/CMakeOutput.log".
See also "/home/zqk/work/project/deep&ros/build/CMakeFiles/CMakeError.log".
```
my torch vision is 0.4.0, is it because my torch  version is too low?
should I upgrade to 1.0?
thanks for your help!

**Comment by ricardo-0x07** [comment]
Im have this issue as well:
mkdir -p build && cd build && cmake .. && make -j4
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Looking for pthread_create
-- Looking for pthread_create - not found
-- Looking for pthread_create in pthreads
-- Looking for pthread_create in pthreads - not found
-- Looking for pthread_create in pthread
-- Looking for pthread_create in pthread - found
-- Found Threads: TRUE  
-- Found CUDA: /usr/local/cuda (found suitable version "10.0", minimum required is "7.0") 
-- Caffe2: CUDA detected: 10.0
-- Caffe2: CUDA nvcc is: /usr/local/cuda/bin/nvcc
-- Caffe2: CUDA toolkit directory: /usr/local/cuda
-- Caffe2: Header version is: 10.0
-- Found CUDNN: /usr/include  
-- Found cuDNN: v5.1.10  (include: /usr/include, library: /usr/lib/x86_64-linux-gnu/libcudnn.so)
CMake Error at /home/clivecadogan/anaconda2/lib/python2.7/site-packages/torch/share/cmake/Caffe2/public/utils.cmake:192 (cuda_select_nvcc_arch_flags):
  Unknown CMake command "cuda_select_nvcc_arch_flags".
Call Stack (most recent call first):
  /home/clivecadogan/anaconda2/lib/python2.7/site-packages/torch/share/cmake/Caffe2/public/cuda.cmake:332 (torch_cuda_get_nvcc_gencode_flag)
  /home/clivecadogan/anaconda2/lib/python2.7/site-packages/torch/share/cmake/Caffe2/Caffe2Config.cmake:88 (include)
  CMakeLists.txt:8 (find_package)


-- Configuring incomplete, errors occurred!
See also "/home/clivecadogan/densepose/build/CMakeFiles/CMakeOutput.log".
See also "/home/clivecadogan/densepose/build/CMakeFiles/CMakeError.log".
Makefile:13: recipe for target 'ops' failed
make: *** [ops] Error 1

**Comment by 3bobo** [comment]
so have you solved this?

**Comment by Yuyiqi1992** [comment]
I met this problem before, upgrading cmake to 3.11.1 is working on my machine.

**Comment by 3bobo** [comment]
@Yuyiqi1992 my cmake version is 3.5.1, and my pytorch versioin is 0.4.0, I'm not sure about which of them is not suitable

**Comment by Yuyiqi1992** [comment]
libtorch is not related to pytorch, it's just using "model.pt" produced by pytorch.

**Comment by 3bobo** [comment]
@Yuyiqi1992 thank you. I upgrade my cmake to 3.11.1 and solved it.

**Comment by dhaji** [comment]
I do not have an issue with the CMake part. However, when I type make I got the following:

davood@davood:~/mnist/build$ cmake -DCMAKE_PREFIX_PATH=~/mnist/libtorch
-- The C compiler identification is GNU 7.3.0
-- The CXX compiler identification is GNU 7.3.0
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Looking for pthread_create
-- Looking for pthread_create - not found
-- Looking for pthread_create in pthreads
-- Looking for pthread_create in pthreads - not found
-- Looking for pthread_create in pthread
-- Looking for pthread_create in pthread - found
-- Found Threads: TRUE  
-- Downloading MNIST dataset
Downloading http://yann.lecun.com/exdb/mnist/train-images-idx3-ubyte.gz ...
0% |################################################################| 100%
Unzipped /home/davood/mnist/build/data/train-images-idx3-ubyte.gz ...
Downloading http://yann.lecun.com/exdb/mnist/train-labels-idx1-ubyte.gz ...
0% |################################################################| 100%
Unzipped /home/davood/mnist/build/data/train-labels-idx1-ubyte.gz ...
Downloading http://yann.lecun.com/exdb/mnist/t10k-images-idx3-ubyte.gz ...
0% |################################################################| 100%
Unzipped /home/davood/mnist/build/data/t10k-images-idx3-ubyte.gz ...
Downloading http://yann.lecun.com/exdb/mnist/t10k-labels-idx1-ubyte.gz ...
0% |################################################################| 100%
Unzipped /home/davood/mnist/build/data/t10k-labels-idx1-ubyte.gz ...
-- Configuring done
-- Generating done
-- Build files have been written to: /home/davood/mnist/build
davood@davood:~/mnist/build$ make
Scanning dependencies of target mnist
[ 50%] Building CXX object CMakeFiles/mnist.dir/mnist.cpp.o
/home/davood/mnist/build/mnist.cpp:106:34: error: ‘torch::data’ has not been declared
 struct Normalize : public torch::data::transforms::TensorTransform<> {
                                  ^~~~
/home/davood/mnist/build/mnist.cpp:106:52: error: expected ‘{’ before ‘TensorTransform’
 struct Normalize : public torch::data::transforms::TensorTransform<> {
                                                    ^~~~~~~~~~~~~~~
/home/davood/mnist/build/mnist.cpp:106:67: error: expected initializer before ‘<’ token
 struct Normalize : public torch::data::transforms::TensorTransform<> {
                                                                   ^
/home/davood/mnist/build/mnist.cpp: In function ‘int main(int, const char**)’:
/home/davood/mnist/build/mnist.cpp:133:14: error: ‘torch::data’ has not been declared
       torch::data::datasets::MNIST(
              ^~~~
/home/davood/mnist/build/mnist.cpp:134:37: error: ‘torch::data’ has not been declared
           options.data_root, torch::data::datasets::MNIST::Mode::kTrain)
                                     ^~~~
/home/davood/mnist/build/mnist.cpp:135:40: error: invalid use of incomplete type ‘struct Normalize’
           .map(Normalize(0.1307, 0.3081))
                                        ^
/home/davood/mnist/build/mnist.cpp:106:8: note: forward declaration of ‘struct Normalize’
 struct Normalize : public torch::data::transforms::TensorTransform<> {
        ^~~~~~~~~
/home/davood/mnist/build/mnist.cpp:136:23: error: ‘torch::data’ has not been declared
           .map(torch::data::transforms::TensorLambda<>(
                       ^~~~
/home/davood/mnist/build/mnist.cpp:136:54: error: expected primary-expression before ‘>’ token
           .map(torch::data::transforms::TensorLambda<>(
                                                      ^
/home/davood/mnist/build/mnist.cpp:138:23: error: ‘torch::data’ has not been declared
           .map(torch::data::transforms::Stack<>());
                       ^~~~
/home/davood/mnist/build/mnist.cpp:138:47: error: expected primary-expression before ‘>’ token
           .map(torch::data::transforms::Stack<>());
                                               ^
/home/davood/mnist/build/mnist.cpp:138:49: error: expected primary-expression before ‘)’ token
           .map(torch::data::transforms::Stack<>());
                                                 ^
/home/davood/mnist/build/mnist.cpp:141:30: error: ‘torch::data’ has not been declared
   auto train_loader = torch::data::make_data_loader(
                              ^~~~
/home/davood/mnist/build/mnist.cpp:144:29: error: ‘torch::data’ has not been declared
   auto test_loader = torch::data::make_data_loader(
                             ^~~~
/home/davood/mnist/build/mnist.cpp:145:14: error: ‘torch::data’ has not been declared
       torch::data::datasets::MNIST(
              ^~~~
/home/davood/mnist/build/mnist.cpp:146:37: error: ‘torch::data’ has not been declared
           options.data_root, torch::data::datasets::MNIST::Mode::kTest)
                                     ^~~~
/home/davood/mnist/build/mnist.cpp:147:40: error: invalid use of incomplete type ‘struct Normalize’
           .map(Normalize(0.1307, 0.3081))
                                        ^
/home/davood/mnist/build/mnist.cpp:106:8: note: forward declaration of ‘struct Normalize’
 struct Normalize : public torch::data::transforms::TensorTransform<> {
        ^~~~~~~~~
/home/davood/mnist/build/mnist.cpp:148:23: error: ‘torch::data’ has not been declared
           .map(torch::data::transforms::TensorLambda<>(
                       ^~~~
/home/davood/mnist/build/mnist.cpp:148:54: error: expected primary-expression before ‘>’ token
           .map(torch::data::transforms::TensorLambda<>(
                                                      ^
/home/davood/mnist/build/mnist.cpp:150:23: error: ‘torch::data’ has not been declared
           .map(torch::data::transforms::Stack<>()),
                       ^~~~
/home/davood/mnist/build/mnist.cpp:150:47: error: expected primary-expression before ‘>’ token
           .map(torch::data::transforms::Stack<>()),
                                               ^
/home/davood/mnist/build/mnist.cpp:150:49: error: expected primary-expression before ‘)’ token
           .map(torch::data::transforms::Stack<>()),
                                                 ^
CMakeFiles/mnist.dir/build.make:62: recipe for target 'CMakeFiles/mnist.dir/mnist.cpp.o' failed
make[2]: *** [CMakeFiles/mnist.dir/mnist.cpp.o] Error 1
CMakeFiles/Makefile2:67: recipe for target 'CMakeFiles/mnist.dir/all' failed
make[1]: *** [CMakeFiles/mnist.dir/all] Error 2
Makefile:83: recipe for target 'all' failed
make: *** [all] Error 2
